examples/stream/stream.cpp

```diff
@@ -13,7 +13,57 @@
 #include <thread>
 #include <vector>
 #include <fstream>
+#include <ctime>
+
+class SimpleWavWriter {
+private:
+    std::ofstream file;
+    int32_t dataSize = 0;
+
+public:
+    SimpleWavWriter(const std::string &filename, int sampleRate, int bitsPerSample, int channels) {
+        file.open(filename, std::ios::binary);
+
+        file.write("RIFF", 4);
+        file.write("\0\0\0\0", 4);    // Placeholder for file size
+        file.write("WAVE", 4);
+        file.write("fmt ", 4);
+
+        int32_t subChunkSize = 16;
+        int16_t audioFormat = 1;      // PCM format
+        int32_t byteRate = sampleRate * channels * bitsPerSample / 8;
+        int16_t blockAlign = channels * bitsPerSample / 8;
+
+        file.write(reinterpret_cast<char *>(&subChunkSize), 4);
+        file.write(reinterpret_cast<char *>(&audioFormat), 2);
+        file.write(reinterpret_cast<char *>(&channels), 2);
+        file.write(reinterpret_cast<char *>(&sampleRate), 4);
+        file.write(reinterpret_cast<char *>(&byteRate), 4);
+        file.write(reinterpret_cast<char *>(&blockAlign), 2);
+        file.write(reinterpret_cast<char *>(&bitsPerSample), 2);
+        file.write("data", 4);
+        file.write("\0\0\0\0", 4);    // Placeholder for data size
+    }
+
+    void writeData(const float *data, size_t length) {
+        for (size_t i = 0; i < length; ++i) {
+            int16_t intSample = static_cast<int16_t>(data[i] * 32767);
+            file.write(reinterpret_cast<char *>(&intSample), sizeof(int16_t));
+            dataSize += sizeof(int16_t);
+        }
```

Comment:
```suggestion
    void writeData(const float *data, size_t length) {
        for (size_t i = 0; i < length; ++i) {
            int16_t intSample = static_cast<int16_t>(data[i] * 32767);
            file.write(reinterpret_cast<char *>(&intSample), sizeof(int16_t));
            dataSize += sizeof(int16_t);
        }
        if (file.is_open()) {
            file.seekp(4, std::ios::beg);
            int32_t fileSize = 36 + dataSize;
            file.write(reinterpret_cast<char *>(&fileSize), 4);
            file.seekp(40, std::ios::beg);
            file.write(reinterpret_cast<char *>(&dataSize), 4);
            file.seekp(0, std::ios::end);
        }
    }
```

Reply:
There's a bug here. Although I understand that you intend to write the audio data size information when this class is destructed, this can pose problems. For instance, in the event of a sudden system crash, or if I terminate the program using `Ctrl+C`, the program will be killed instantly, leaving no time to write the audio data size information. Consequently, the audio would be corrupted and unplayable.